### PR TITLE
Unit tests for std::iter::Sum for fieldElement

### PR DESCRIPTION
**Pull request by tcoratger**
# Add test for `std::iter::Sum` for `FieldElement`

## Description
As a response to [#154](https://github.com/lambdaclass/lambdaworks/issues/154), unit testing functions for `std::iter::Sum` applied to a `FieldElement` (`U64TestField` for the test):
- `test_std_iter_sum_field_element` arbitrary length vector test
- `test_std_iter_sum_field_element_zero_length` zero length vector test

## Type of change

- [x] Testing

## Checklist
- [x] Linked to Github Issue
- [x] Unit tests added

### CODE REVIEW

**Comment by jrchatruc** [comment]
Hi, thanks for the PR! It's looking good, just remove the vscode `settings.json` file.

**Comment by tcoratger** [comment]
@jrchatruc Sorry, `vscode` settings are now removed.

**Review by jrchatruc — APPROVED** [review]

**Comment by jrchatruc** [comment]
> @jrchatruc Sorry, `vscode` settings are now removed.

Awesome, just fix the conflicts and it's ready to merge

**Comment by jrchatruc** [comment]
> @jrchatruc Done, seems like there is an issue on an element external to the PR, in the `representative` function inside the `math/src/field/fields/p448_goldilocks_prime_field.rs` file.

Yeah you're right, I fixed it on [this PR](https://github.com/lambdaclass/lambdaworks/pull/169), can you bring in the latest changes?

**Comment by tcoratger** [comment]
@jrchatruc Done :)

File: math/src/field/element.rs [hunk 1]
```diff
@@ -368,3 +368,42 @@ where
...
+    #[test]
+    fn test_std_iter_sum_field_element() {
+        let n = 164;
+        const MODULUS: u64 = 15;
+        assert_eq!(
+            (0..n)
+                .map(|x| { FieldElement::<U64TestField<MODULUS>>::from(x) })
+                .collect::<Vec<_>>()
+                .iter()
+                .cloned()
```
Comment (ilitteri) [inline math/src/field/element.rs:387]: These lines are not needed.

File: math/src/field/element.rs [hunk 2]
```diff
@@ -368,3 +368,42 @@ where
...
+    #[test]
+    fn test_std_iter_sum_field_element_zero_length() {
+        let n = 0;
+        const MODULUS: u64 = 15;
+        assert_eq!(
+            (0..n)
+                .map(|x| { FieldElement::<U64TestField<MODULUS>>::from(x) })
+                .collect::<Vec<_>>()
+                .iter()
+                .cloned()
```
Comment (ilitteri) [inline math/src/field/element.rs:403]: Ditto.

File: math/src/field/element.rs (hunk 1 above)
Comment (tcoratger) [reply math/src/field/element.rs]: @ilitteri Clean up is done

File: math/src/field/element.rs (hunk 2 above)
Comment (tcoratger) [reply math/src/field/element.rs]: @ilitteri Same, clean up is done

**Review by ilitteri — APPROVED** [review]
LGTM!

**Review by entropidelic — APPROVED** [review]

**Pull request merged**